File: 26. Remove Duplicates from Sorted Array/no_extra.cpp
class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        int res = 1;
        for (int i = 0; i < nums.size() - 1; i++) {
            if (nums[i] != nums[i+1]) {
                nums[res] = nums[i+1];
                res += 1;
            }
        }
        return res;
    }
};

File: 26. Remove Duplicates from Sorted Array/no_extra.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        res = 1
        for i in range(len(nums) - 1):
            if nums[i] != nums[i+1]:
                nums[res] = nums[i+1]
                res += 1
        return res